### PR TITLE
Add backend checks results lambda.

### PR DESCRIPTION
**Pull request by MancunianSam**
This takes the output in S3 from the map function and converts it to json in a form more easily consumed by the other lambdas.

See the README for more information.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+jobs:
+  pre-deploy:
+    uses: nationalarchives/tdr-github-actions/.github/workflows/lambda_build.yml@main
+    with:
+      repo-name: tdr-file-upload-data
```
Comment (TomJKing) [inline .github/workflows/build.yml]: Is that the correct `repo-name` value?

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+name: TDR Run File Upload Data Tests
```
Comment (TomJKing) [inline .github/workflows/test.yml]: Is that the correct `name` value?

**Review by TomJKing — COMMENTED** [review]
Looks good. Couple of minor comments.

File: .github/workflows/build.yml (hunk 1 above)
Comment (MancunianSam) [reply .github/workflows/build.yml]: It's not, copy and paste errors strike again.

File: .github/workflows/test.yml (hunk 2 above)
Comment (MancunianSam) [reply .github/workflows/test.yml]: No, I've changed this as well.

**Review by TomJKing — APPROVED** [review]
Looks good. Thanks

**Pull request merged**